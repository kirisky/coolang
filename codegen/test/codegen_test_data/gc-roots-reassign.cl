class A {
 foo:Int;
};

class B inherits A {
 bar:Int;
};

class Killer {
 foo:Int;
};

class Main inherits IO {
  main() : Object { {
    let
      a:A <- new A,
    in {
      a <- new B;
	  out_string("A should die now\n");
	  new Killer;
    };
	out_string("B should die now\n");
	new Killer;
  } };
};
