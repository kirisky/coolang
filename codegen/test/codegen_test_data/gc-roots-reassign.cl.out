Allocated an object of type: Main
Allocated an object of type: A
Allocated an object of type: B
A should die now
Freeing an object of type: A
Allocated an object of type: Killer
B should die now
Freeing an object of type: Killer
Freeing an object of type: B
Allocated an object of type: Killer
